src/redis-loader.ts

```diff
@@ -0,0 +1,437 @@
+  migrate: (...args: any[]) => any;
+  migrateBuffer: (...args: any[]) => any;
+  dump: (key: string) => Promise<string>;
+  dumpBuffer: (key: string) => Promise<string>;
+  object: (subcommand: string, ...args: any[]) => any;
+  objectBuffer: (subcommand: string, ...args: any[]) => any;
+  client: (...args: any[]) => any;
+  clientBuffer: (...args: any[]) => any;
+  eval: (...args: any[]) => any;
+  evalBuffer: (...args: any[]) => any;
+  evalsha: (...args: any[]) => any;
+  evalshaBuffer: (...args: any[]) => any;
+  script: (...args: any[]) => any;
+  scriptBuffer: (...args: any[]) => any;
+  quit: () => Promise<string>;
+  quitBuffer: () => Promise<string>;
+  scan: (cursor: number, ...args: any[]) => any;
+  scanBuffer: (cursor: number, ...args: any[]) => any;
+  hscan: (key: string, cursor: number, ...args: any[]) => any;
+  hscanBuffer: (key: string, cursor: number, ...args: any[]) => any;
+  zscan: (key: string, cursor: number, ...args: any[]) => any;
+  zscanBuffer: (key: string, cursor: number, ...args: any[]) => any;
+  pfmerge: (destkey: string, ...sourcekeys: string[]) => any;
+  pfmergeBuffer: (destkey: string, ...sourcekeys: string[]) => any;
+  pfadd: (key: string, ...elements: string[]) => any;
+  pfaddBuffer: (key: string, ...elements: string[]) => any;
+  pfcount: (...keys: string[]) => any;
+  pfcountBuffer: (...keys: string[]) => any;
+  scanStream: (options?: ScanStreamOptions) => ScanStream;
+  scanStreamBuffer: (options?: ScanStreamOptions) => ScanStream;
+  sscanStream: (options?: ScanStreamOptions) => ScanStream;
+  sscanStreamBuffer: (options?: ScanStreamOptions) => ScanStream;
+  hscanStream: (key: string, options?: ScanStreamOptions) => ScanStream;
+  hscanStreamBuffer: (key: string, options?: ScanStreamOptions) => ScanStream;
+  zscanStream: (key: string, options?: ScanStreamOptions) => ScanStream;
+  zscanStreamBuffer: (key: string, options?: ScanStreamOptions) => ScanStream;
+
+  // The proxied event emitter methods
+  addListener: (event: string | symbol, listener: Function) => this;
+  on: (event: string | symbol, listener: Function) => this;
+  once: (event: string | symbol, listener: Function) => this;
+  prependListener: (event: string | symbol, listener: Function) => this;
+  prependOnceListener: (event: string | symbol, listener: Function) => this;
+  removeListener: (event: string | symbol, listener: Function) => this;
+  removeAllListeners: (event?: string | symbol) => this;
+  setMaxListeners: (n: number) => this;
+  getMaxListeners: () => number;
+  listeners: (event: string | symbol) => Function[];
+  emit: (event: string | symbol, ...args: any[]) => boolean;
+  eventNames: () => (string | symbol)[];
+  listenerCount: (type: string | symbol) => number;
+}
+
+// Dynamically create the redis functions on our prototype
+// https://github.com/luin/ioredis/blob/58dc63006fcc98f7a13f363f920856afa887801e/lib/commander.js#L28
+const redisCommands = new Set(redisCommandList)
```

Comment:
can you move this content to the top of the file? hard to dig down through all those method signatures